pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +99,268 @@ func (builder *Builder) validate() (bool, error) {
 
```

Comment:
Please add a block for check errorMsg:
```
	if builder.errorMsg != "" {
		glog.V(100).Infof("The %s builder has error message: %s", resourceCRD, builder.errorMsg)

		return false, fmt.Errorf(builder.errorMsg)
	}
```

Reply:
Also, please move the validate() method to the end of the file since it is private